build-python.sh

```diff
@@ -342,7 +346,7 @@ if test "${BUILD_SDIST}" = true; then
     python -m build \
         --sdist \
         --outdir ../dist \
-        ${BUILD_ARGS} \
+        ${BUILD_ARGS:+ -o "$BUILD_ARGS"} \
```

Comment:
I don't understand this change, can you please explain it?

Reply:
@vnherdeiro are you still interested in this work and available to help? If not, could you please let us know so we can continue it in other PRs?

Reply:
Thanks for that.

The `${BUILD_ARGS:+"${BUILD_ARGS}"}` form is not what we want for this script. This warning is about quoting to "prevent word splitting" ... but we genuinely **want** word splitting here, so that this string containing multiple command line arguments (separated by spaces) will be passed through to the underlying commands that way, not as a single long string to one positional argument.

And we cannot use arrays because this is `sh`, not `bash`.

I just pushed a commit reverting these changes and just adding `#shellcheck disable` comments.